lib/declare_schema/model/index_definition.rb

```diff
@@ -104,12 +105,21 @@ def primary_key?
         name == PRIMARY_KEY_NAME
       end
 
+      def options
+        @options ||=
+          OPTIONS.each_with_object({}) do |option, result|
+            result[option] = send(option)
+          end.freeze
+      end
+
+      # Unique key for this object. Used for equality checking.
       def to_key
         @to_key ||= [name, *settings].freeze
       end
 
+      # The index settings for this object. Used for equivalence checking. Does not include the name.
       def settings
-        @settings ||= [columns, unique, where].freeze
+        @settings ||= [columns, options.except(:name)].freeze
```

Comment:
This looks to have changed the format that this is returning right?
`unique` and `where` were strings, and `options.except(:name)` is a hash?

Is this expected?

Reply:
Yes, the `settings` just need to be unique. It wasn't necessary to flatten them into strings.

And in the past there had been bugs where an option was added but didn't get added to the settings or key. So I wanted them to track whatever's in `options`.